modules/utxo-lib/test/testutil/tnumber.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Multiply a decimal amount (optionally to a bigint)
+ * Throws error if resulting value is not a safe integer number
```

Comment:
This can be a little more explicit, it's not clear what we're multiplying by. How about "Multiply a decimal amount of coins by 1e8 to get an integer number of satoshis" or similar?

Reply:
> Also, what is unsafe about this method? Or was it just unsafe the way it was implemented previously?

Precise decimal values are commonly not representable exactly in floating point, and should _never_ be used to represent monetary values of any. https://www.exploringbinary.com/why-0-point-1-does-not-exist-in-floating-point/

This type of function papers over that type of error (either the original implementation or my simplified one) by rounding, but should really never ever ever be used in production.